src/main/java/org/assertj/core/internal/Files.java

```diff
@@ -507,6 +507,25 @@ public void assertIsDirectoryNotContaining(AssertionInfo info, File actual, Stri
     assertIsDirectoryNotContaining(info, actual, fileMatcher, format("the '%s' pattern", syntaxAndPattern));
   }
 
+  /**
+   * Get the content of the given {@code File}.
+   *
+   * @param info contains information about the assertion.
+   * @param actual the given file.
+   * @throws AssertionError if the actual {@code File} is {@code null}.
+   * @throws AssertionError if the actual {@code File} can not be read.
+   * @return a String object with the content of the given {@code File}.
+   */
+
+  public String getFileContent(AssertionInfo info, File actual) {
+    assertCanRead(info, actual);
+    try {
+      return new String(readAllBytes(actual.toPath()));
+    } catch (IOException e) {
+      throw failures.failure(info, shouldBeReadable(actual));
```

Comment:
rethrow an `UncheckedIOException` as the file is readable (in term of permissions) but the content led to an I/O error.